#Задание 1

In [1]:
from google.colab import files
uploaded = files.upload()

Saving weatherHistory.csv to weatherHistory.csv


In [0]:
import pandas as pd

In [0]:
weather_hist = pd.read_csv('weatherHistory.csv')

In [0]:
pd.set_option('display.max_columns', 100)
pd.set_option('display.max_rows', 100)

In [5]:
weather_hist.shape

(96453, 12)

In [6]:
weather_hist.head(5)

Formatted Date        Summary Precip Type  Temperature (C)  \
0  2006-04-01 00:00:00.000 +0200  Partly Cloudy        rain         9.472222   
1  2006-04-01 01:00:00.000 +0200  Partly Cloudy        rain         9.355556   
2  2006-04-01 02:00:00.000 +0200  Mostly Cloudy        rain         9.377778   
3  2006-04-01 03:00:00.000 +0200  Partly Cloudy        rain         8.288889   
4  2006-04-01 04:00:00.000 +0200  Mostly Cloudy        rain         8.755556   

   Apparent Temperature (C)  Humidity  Wind Speed (km/h)  \
0                  7.388889      0.89            14.1197   
1                  7.227778      0.86            14.2646   
2                  9.377778      0.89             3.9284   
3                  5.944444      0.83            14.1036   
4                  6.977778      0.83            11.0446   

   Wind Bearing (degrees)  Visibility (km)  Loud Cover  Pressure (millibars)  \
0                   251.0          15.8263         0.0               1015.13   
1                   259.0          15.8263         0.0               1015.63   
2                   204.0          14.9569         0.0               1015.94   
3                   269.0          15.8263         0.0               1016.41   
4                   259.0          15.8263         0.0               1016.51   

                       Daily Summary  
0  Partly cloudy throughout the day.  
1  Partly cloudy throughout the day.  
2  Partly cloudy throughout the day.  
3  Partly cloudy throughout the day.  
4  Partly cloudy throughout the day.

In [0]:
weather_hist.rename({'Formatted Date':'date', 'Temperature (C)':'temp', 'Apparent Temperature (C)':'temp_app',
                    'Precip Type':'weather_type', 'Wind Speed (km/h)':'wind_speed', 'Humidity':'humidity',
                    'Daily Summary':'summary'}, 
                    axis=1, inplace=True)

In [11]:
weather_hist.head(5)

date        Summary weather_type      temp  \
0  2006-04-01 00:00:00.000 +0200  Partly Cloudy         rain  9.472222   
1  2006-04-01 01:00:00.000 +0200  Partly Cloudy         rain  9.355556   
2  2006-04-01 02:00:00.000 +0200  Mostly Cloudy         rain  9.377778   
3  2006-04-01 03:00:00.000 +0200  Partly Cloudy         rain  8.288889   
4  2006-04-01 04:00:00.000 +0200  Mostly Cloudy         rain  8.755556   

   temp_app  humidity  wind_speed  Wind Bearing (degrees)  Visibility (km)  \
0  7.388889      0.89     14.1197                   251.0          15.8263   
1  7.227778      0.86     14.2646                   259.0          15.8263   
2  9.377778      0.89      3.9284                   204.0          14.9569   
3  5.944444      0.83     14.1036                   269.0          15.8263   
4  6.977778      0.83     11.0446                   259.0          15.8263   

   Pressure (millibars)                            summary  
0               1015.13  Partly cloudy throughout the day.  
1               1015.63  Partly cloudy throughout the day.  
2               1015.94  Partly cloudy throughout the day.  
3               1016.41  Partly cloudy throughout the day.  
4               1016.51  Partly cloudy throughout the day.

In [0]:
weather_hist.drop(['Loud Cover'], axis=1, inplace=True)

In [13]:
weather_hist['temp'].min(), weather_hist['temp'].max()

(-21.822222222222226, 39.90555555555555)

In [14]:
weather_hist['wind_speed'].min(), weather_hist['wind_speed'].max()

(0.0, 63.8526)

Задания

In [15]:
weather_hist.date

0        2006-04-01 00:00:00.000 +0200
1        2006-04-01 01:00:00.000 +0200
2        2006-04-01 02:00:00.000 +0200
3        2006-04-01 03:00:00.000 +0200
4        2006-04-01 04:00:00.000 +0200
                     ...              
96448    2016-09-09 19:00:00.000 +0200
96449    2016-09-09 20:00:00.000 +0200
96450    2016-09-09 21:00:00.000 +0200
96451    2016-09-09 22:00:00.000 +0200
96452    2016-09-09 23:00:00.000 +0200
Name: date, Length: 96453, dtype: object

In [16]:
weather_hist.head(30)

date        Summary weather_type       temp  \
0   2006-04-01 00:00:00.000 +0200  Partly Cloudy         rain   9.472222   
1   2006-04-01 01:00:00.000 +0200  Partly Cloudy         rain   9.355556   
2   2006-04-01 02:00:00.000 +0200  Mostly Cloudy         rain   9.377778   
3   2006-04-01 03:00:00.000 +0200  Partly Cloudy         rain   8.288889   
4   2006-04-01 04:00:00.000 +0200  Mostly Cloudy         rain   8.755556   
5   2006-04-01 05:00:00.000 +0200  Partly Cloudy         rain   9.222222   
6   2006-04-01 06:00:00.000 +0200  Partly Cloudy         rain   7.733333   
7   2006-04-01 07:00:00.000 +0200  Partly Cloudy         rain   8.772222   
8   2006-04-01 08:00:00.000 +0200  Partly Cloudy         rain  10.822222   
9   2006-04-01 09:00:00.000 +0200  Partly Cloudy         rain  13.772222   
10  2006-04-01 10:00:00.000 +0200  Partly Cloudy         rain  16.016667   
11  2006-04-01 11:00:00.000 +0200  Partly Cloudy         rain  17.144444   
12  2006-04-01 12:00:00.000 +0200  Partly Cloudy         rain  17.800000   
13  2006-04-01 13:00:00.000 +0200  Partly Cloudy         rain  17.333333   
14  2006-04-01 14:00:00.000 +0200  Partly Cloudy         rain  18.877778   
15  2006-04-01 15:00:00.000 +0200  Partly Cloudy         rain  18.911111   
16  2006-04-01 16:00:00.000 +0200  Partly Cloudy         rain  15.388889   
17  2006-04-01 17:00:00.000 +0200  Mostly Cloudy         rain  15.550000   
18  2006-04-01 18:00:00.000 +0200  Mostly Cloudy         rain  14.255556   
19  2006-04-01 19:00:00.000 +0200  Mostly Cloudy         rain  13.144444   
20  2006-04-01 20:00:00.000 +0200  Mostly Cloudy         rain  11.550000   
21  2006-04-01 21:00:00.000 +0200  Mostly Cloudy         rain  11.183333   
22  2006-04-01 22:00:00.000 +0200  Partly Cloudy         rain  10.116667   
23  2006-04-01 23:00:00.000 +0200  Mostly Cloudy         rain  10.200000   
24  2006-04-10 00:00:00.000 +0200  Partly Cloudy         rain  10.422222   
25  2006-04-10 01:00:00.000 +0200  Partly Cloudy         rain   9.911111   
26  2006-04-10 02:00:00.000 +0200  Mostly Cloudy         rain  11.183333   
27  2006-04-10 03:00:00.000 +0200  Partly Cloudy         rain   7.155556   
28  2006-04-10 04:00:00.000 +0200  Partly Cloudy         rain   6.111111   
29  2006-04-10 05:00:00.000 +0200  Partly Cloudy         rain   6.788889   

     temp_app  humidity  wind_speed  Wind Bearing (degrees)  Visibility (km)  \
0    7.388889      0.89     14.1197                   251.0          15.8263   
1    7.227778      0.86     14.2646                   259.0          15.8263   
2    9.377778      0.89      3.9284                   204.0          14.9569   
3    5.944444      0.83     14.1036                   269.0          15.8263   
4    6.977778      0.83     11.0446                   259.0          15.8263   
5    7.111111      0.85     13.9587                   258.0          14.9569   
6    5.522222      0.95     12.3648                   259.0           9.9820   
7    6.527778      0.89     14.1519                   260.0           9.9820   
8   10.822222      0.82     11.3183                   259.0           9.9820   
9   13.772222      0.72     12.5258                   279.0           9.9820   
10  16.016667      0.67     17.5651                   290.0          11.2056   
11  17.144444      0.54     19.7869                   316.0          11.4471   
12  17.800000      0.55     21.9443                   281.0          11.2700   
13  17.333333      0.51     20.6885                   289.0          11.2700   
14  18.877778      0.47     15.3755                   262.0          11.4471   
15  18.911111      0.46     10.4006                   288.0          11.2700   
16  15.388889      0.60     14.4095                   251.0          11.2700   
17  15.550000      0.63     11.1573                   230.0          11.4471   
18  14.255556      0.69      8.5169                   163.0          11.2056   
19  13.144444      0.70      7.6314                   139.0          11.

In [0]:
cold_days = weather_hist.groupby(['temp', 'date']).agg(['min'])

In [18]:
cold_days.head(1)

Summary weather_type   temp_app  \
                                             min          min        min   
temp       date                                                            
-21.822222 2012-02-10 07:00:00.000 +0100   Foggy         snow -21.822222   

                                         humidity wind_speed  \
                                              min        min   
temp       date                                                
-21.822222 2012-02-10 07:00:00.000 +0100      0.8     3.0751   

                                         Wind Bearing (degrees)  \
                                                            min   
temp       date                                                   
-21.822222 2012-02-10 07:00:00.000 +0100                  323.0   

                                         Visibility (km) Pressure (millibars)  \
                                                     min                  min   
temp       date                                                                 
-21.822222 2012-02-10 07:00:00.000 +0100          1.3685              1033.66   

                                                                                    summary  
                                                                                        min  
temp       date                                                                              
-21.822222 2012-02-10 07:00:00.000 +0100  Foggy starting in the morning continuing until...

Самый холодный день - 2012-02-10

-21.822222



In [0]:
windy_days = weather_hist.groupby(['wind_speed', 'date']).agg(['min'])

In [20]:
windy_days.tail(1)

Summary  \
                                                                          min   
wind_speed date                                                                 
63.8526    2007-01-29 13:00:00.000 +0100  Dangerously Windy and Partly Cloudy   

                                         weather_type      temp  temp_app  \
                                                  min       min       min   
wind_speed date                                                             
63.8526    2007-01-29 13:00:00.000 +0100         rain  8.944444  3.483333   

                                         humidity Wind Bearing (degrees)  \
                                              min                    min   
wind_speed date                                                            
63.8526    2007-01-29 13:00:00.000 +0100     0.49                  307.0   

                                         Visibility (km) Pressure (millibars)  \
                                                     min                  min   
wind_speed date                                                                 
63.8526    2007-01-29 13:00:00.000 +0100         11.4471              1009.05   

                                                                                    summary  
                                                                                        min  
wind_speed date                                                                              
63.8526    2007-01-29 13:00:00.000 +0100  Mostly cloudy throughout the day and windy sta...

Самый ветренный день - 2007-01-29 

63.8526 

P. S.
Не знаю почему, но когда использовал такую же команду, как для определения температурного минимума (только поменял на "max"), выдавало 0.0. Так что пришлось выкручиваться. Буду благодарен за комментарий на этот счёт. 



In [0]:
weather_hist['date'] = pd.to_datetime(weather_hist.date, utc=True)



In [22]:
weather_hist.dtypes

date                      datetime64[ns, UTC]
Summary                                object
weather_type                           object
temp                                  float64
temp_app                              float64
humidity                              float64
wind_speed                            float64
Wind Bearing (degrees)                float64
Visibility (km)                       float64
Pressure (millibars)                  float64
summary                                object
dtype: object

In [0]:
summer_start = pd.to_datetime('2014-06-01', utc=True)
summer_end = pd.to_datetime('2014-08-31', utc=True)

In [24]:
summer_end

Timestamp('2014-08-31 00:00:00+0000', tz='UTC')

In [0]:
r1 = weather_hist.loc[summer_start <= weather_hist.date, :]


In [0]:
r2 = r1.loc[weather_hist.date <= summer_end, :]

In [27]:
r2.head(5)

date        Summary weather_type       temp  \
70872 2014-07-31 22:00:00+00:00       Overcast         rain  20.422222   
70873 2014-07-31 23:00:00+00:00       Overcast         rain  19.983333   
70874 2014-08-01 00:00:00+00:00       Overcast         rain  19.850000   
70875 2014-08-01 01:00:00+00:00  Mostly Cloudy         rain  19.400000   
70876 2014-08-01 02:00:00+00:00  Partly Cloudy         rain  18.888889   

        temp_app  humidity  wind_speed  Wind Bearing (degrees)  \
70872  20.422222      0.90      5.6350                   130.0   
70873  19.983333      0.93      6.1824                   358.0   
70874  19.850000      0.93      4.8944                   313.0   
70875  19.400000      0.93     11.2700                   330.0   
70876  18.888889      0.93      2.5760                   128.0   

       Visibility (km)  Pressure (millibars)  \
70872           8.3720               1010.20   
70873           9.9820               1009.64   
70874          11.2539               1009.40   
70875          14.9086               1009.63   
70876          11.9784               1009.89   

                                 summary  
70872  Partly cloudy throughout the day.  
70873  Partly cloudy throughout the day.  
70874  Partly cloudy throughout the day.  
70875  Partly cloudy throughout the day.  
70876  Partly cloudy throughout the day.

In [28]:

r2.weather_type.value_counts()

rain    2182
Name: weather_type, dtype: int64

Разделив на 24 (количестыо замеров температуры в сутки) получаем 91 день (округляем в большую сторону, так как если хотя бы один из замеров показал "дождь", день уже можно считать дождливым) 

# Cамый ветрянный месяц, самый влажный год  и неделя с самой переменчивой погодой

In [29]:
weather_hist['Year'] = weather_hist.date.dt.year
weather_hist['Month'] = weather_hist.date.dt.month
weather_hist['Week'] = weather_hist.date.dt.week 

weather_hist.head(5)

date        Summary weather_type      temp  temp_app  \
0 2006-03-31 22:00:00+00:00  Partly Cloudy         rain  9.472222  7.388889   
1 2006-03-31 23:00:00+00:00  Partly Cloudy         rain  9.355556  7.227778   
2 2006-04-01 00:00:00+00:00  Mostly Cloudy         rain  9.377778  9.377778   
3 2006-04-01 01:00:00+00:00  Partly Cloudy         rain  8.288889  5.944444   
4 2006-04-01 02:00:00+00:00  Mostly Cloudy         rain  8.755556  6.977778   

   humidity  wind_speed  Wind Bearing (degrees)  Visibility (km)  \
0      0.89     14.1197                   251.0          15.8263   
1      0.86     14.2646                   259.0          15.8263   
2      0.89      3.9284                   204.0          14.9569   
3      0.83     14.1036                   269.0          15.8263   
4      0.83     11.0446                   259.0          15.8263   

   Pressure (millibars)                            summary  Year  Month  Week  
0               1015.13  Partly cloudy throughout the day.  2006      3    13  
1               1015.63  Partly cloudy throughout the day.  2006      3    13  
2               1015.94  Partly cloudy throughout the day.  2006      4    13  
3               1016.41  Partly cloudy throughout the day.  2006      4    13  
4               1016.51  Partly cloudy throughout the day.  2006      4    13

In [0]:
years = weather_hist.groupby('Year')

#years.head(5)

In [0]:
monthes = weather_hist.groupby('Month')


#monthes.head(5)

In [0]:
weeks = weather_hist.groupby('Week')

#weeks.head(5)

In [0]:
monthes.dtypes

date Summary weather_type     temp temp_app humidity  \
Month                                                                        
1      datetime64[ns, UTC]  object       object  float64  float64  float64   
2      datetime64[ns, UTC]  object       object  float64  float64  float64   
3      datetime64[ns, UTC]  object       object  float64  float64  float64   
4      datetime64[ns, UTC]  object       object  float64  float64  float64   
5      datetime64[ns, UTC]  object       object  float64  float64  float64   
6      datetime64[ns, UTC]  object       object  float64  float64  float64   
7      datetime64[ns, UTC]  object       object  float64  float64  float64   
8      datetime64[ns, UTC]  object       object  float64  float64  float64   
9      datetime64[ns, UTC]  object       object  float64  float64  float64   
10     datetime64[ns, UTC]  object       object  float64  float64  float64   
11     datetime64[ns, UTC]  object       object  float64  float64  float64   
12     datetime64[ns, UTC]  object       object  float64  float64  float64   

      wind_speed Wind Bearing (degrees) Visibility (km) Pressure (millibars)  \
Month                                                                          
1        float64                float64         float64              float64   
2        float64                float64         float64              float64   
3        float64                float64         float64              float64   
4        float64                float64         float64              float64   
5        float64                float64         float64              float64   
6        float64                float64         float64              float64   
7        float64                float64         float64              float64   
8        float64                float64         float64              float64   
9        float64                float64         float64              float64   
10       float64                float64         float64              float64   
11       float64                float64         float64              float64   
12       float64                float64         float64              float64   

      summary   Year   Week  
Month                        
1      object  int64  int64  
2      object  int64  int64  
3      object  int64  int64  
4      object  int64  int64  
5      object  int64  int64  
6      object  int64  int64  
7      object  int64  int64  
8      object  int64  int64  
9      object  int64  int64  
10     object  int64  int64  
11     object  int64  int64  
12     object  int64  int64

In [33]:
monthes.sum(level='wind_speed')

Summary           temp  \
Month                                                                     
1      OvercastMostly CloudyMostly CloudyOvercastMost...    6675.505556   
2      OvercastOvercastOvercastOvercastOvercastOverca...   16173.588889   
3      Partly CloudyPartly CloudyOvercastOvercastOver...   56592.166667   
4      Mostly CloudyPartly CloudyMostly CloudyPartly ...  101132.066667   
5      Partly CloudyPartly CloudyOvercastMostly Cloud...  138100.538889   
6      Mostly CloudyMostly CloudyMostly CloudyClearPa...  164169.800000   
7      Partly CloudyClearPartly CloudyPartly CloudyPa...  187880.444444   
8      Mostly CloudyPartly CloudyPartly CloudyPartly ...  183359.611111   
9      ClearClearClearClearPartly CloudyPartly Cloudy...  138627.388889   
10     Mostly CloudyPartly CloudyClearClearClearFoggy...   92758.427778   
11     OvercastMostly CloudyMostly CloudyMostly Cloud...   52173.388889   
12     Mostly CloudyOvercastOvercastOvercastOvercastO...   13299.705556   

            temp_app  humidity   wind_speed  Wind Bearing (degrees)  \
Month                                                                 
1      -15866.816667   6962.11   94245.8419               1476890.0   
2       -4149.300000   6071.27   90917.6016               1408705.0   
3       41595.855556   5752.12  109795.1092               1527402.0   
4       96293.633333   5077.12   94059.0175               1476572.0   
5      136945.044444   5660.46   89640.8716               1539765.0   
6      164382.883333   5436.43   76289.0289               1606215.0   
7      187979.788889   5232.51   78841.3136               1660228.0   
8      182747.833333   5214.54   73401.7354               1602293.0   
9      137727.661111   5458.17   76112.8144               1485119.0   
10      87443.361111   6338.04   81909.5228               1448607.0   
11      39760.422222   6557.10   86707.7897               1371557.0   
12      -7860.266667   7123.34   90798.0269               1482475.0   

       Visibility (km)  Pressure (millibars)  \
Month                                          
1           64128.3930            8235110.90   
2           65164.7178            7493297.14   
3           89387.5381            8196807.95   
4           93337.1257            7999161.62   
5           97305.5503            8211637.12   
6           94992.4794            7935995.09   
7           99714.9797            8251052.71   
8          102224.9536            8220070.61   
9           91859.4355            7926616.63   
10          79629.9721            8209366.21   
11          64880.0215            8018367.42   
12          55405.3647            8067634.25   

                                                 summary      Year    Week  
Month                                                                       
1      Mostly cloudy until night.Mostly cloudy throug...  16458024   34824  
2      Mostly cloudy until night.Mostly cloudy until ...  15010176   51864  
3      Partly cloudy throughout the day.Partly cloudy...  16458024   91512  
4      Partly cloudy throughout the day.Partly cloudy...  15927120  123072  
5      Mostly cloudy until night.Mostly cloudy until ...  16458024  162816  
6      Mostly cloudy throughout the day.Mostly cloudy...  15927120  192096  
7      Partly cloudy starting overnight.Partly cloudy...  16451982  234059  
8      Partly cloudy starting overnight.Partly cloudy...  16506264  271126  
9      Foggy in the morning.Foggy in the morning.Part...  15927120  296184  
10     Mostly cloudy until night.Foggy in the morning...  16458024  341712  
11     Foggy in the morning.Mostly cloudy until night...  15927120  365184  
12     Foggy in the morning.Foggy in the morning.Fogg...  16458013  400560

In [34]:
monthes.sum(level='wind_speed').agg(['max'])

Summary           temp  \
max  Partly CloudyPartly CloudyOvercastOvercastOver...  187880.444444   

          temp_app  humidity   wind_speed  Wind Bearing (degrees)  \
max  187979.788889   7123.34  109795.1092               1660228.0   

     Visibility (km)  Pressure (millibars)  \
max      102224.9536            8251052.71   

                                               summary      Year    Week  
max  Partly cloudy throughout the day.Partly cloudy...  16506264  400560

Сопоставив таблицы, делаем вывод что март - самый ветрянный месяц. 

(взяты общие показатели месяцев за все года - то есть март, в среднем, самый ветрянный месяц) 

In [35]:
years.dtypes.head(5)

date Summary weather_type     temp temp_app humidity  \
Year                                                                        
2005  datetime64[ns, UTC]  object       object  float64  float64  float64   
2006  datetime64[ns, UTC]  object       object  float64  float64  float64   
2007  datetime64[ns, UTC]  object       object  float64  float64  float64   
2008  datetime64[ns, UTC]  object       object  float64  float64  float64   
2009  datetime64[ns, UTC]  object       object  float64  float64  float64   

     wind_speed Wind Bearing (degrees) Visibility (km) Pressure (millibars)  \
Year                                                                          
2005    float64                float64         float64              float64   
2006    float64                float64         float64              float64   
2007    float64                float64         float64              float64   
2008    float64                float64         float64              float64   
2009    float64                float64         float64              float64   

     summary  Month   Week  
Year                        
2005  object  int64  int64  
2006  object  int64  int64  
2007  object  int64  int64  
2008  object  int64  int64  
2009  object  int64  int64

In [38]:
years.sum(level='humidity')


Summary           temp  \
Year                                                                     
2005                                      Partly Cloudy       0.577778   
2006  Partly CloudyPartly CloudyMostly CloudyPartly ...   98245.366667   
2007  Partly CloudyPartly CloudyClearClearClearPartl...  106299.766667   
2008  ClearClearMostly CloudyClearClearClearPartly C...  106829.416667   
2009  OvercastOvercastPartly CloudyOvercastOvercastO...  107482.416667   
2010  ClearClearClearClearClearClearClearClearClearC...   98382.350000   
2011  ClearClearPartly CloudyClearClearPartly Cloudy...  100958.422222   
2012  Breezy and OvercastMostly CloudyBreezy and Mos...  105292.261111   
2013  Mostly CloudyPartly CloudyPartly CloudyMostly ...  104603.305556   
2014  Mostly CloudyMostly CloudyMostly CloudyMostly ...  109709.688889   
2015  OvercastBreezy and OvercastOvercastOvercastOve...  107853.894444   
2016  Partly CloudyPartly CloudyPartly CloudyMostly ...  105285.166667   

           temp_app  humidity   wind_speed  Wind Bearing (degrees)  \
Year                                                                 
2005      -4.050000      0.89      17.1143                   140.0   
2006   88985.738889   6721.98   89256.7578               1658616.0   
2007   96911.755556   6041.31   94819.6620               1700372.0   
2008   97090.672222   6159.68   99295.8484               1702875.0   
2009   96980.783333   6195.48  100804.1769               1580985.0   
2010   88692.066667   6999.63   96749.9876               1687220.0   
2011   92720.133333   6447.43   86711.7020               1658570.0   
2012   94455.377778   6056.68   98949.5213               1649169.0   
2013   94831.461111   6606.77   96097.5190               1645676.0   
2014  101823.322222   6555.24   91959.6741               1560857.0   
2015   99349.666667   6415.42   94043.7708               1612581.0   
2016   95163.172222   6682.70   94012.9393               1628767.0   

      Visibility (km)  Pressure (millibars)  \
Year                                          
2005           9.9820               1016.66   
2006       85473.7730            8694698.46   
2007       91175.7651            8774361.25   
2008       90184.2949            8851939.74   
2009       87619.1624            8782820.28   
2010       80288.2367            8826298.82   
2011       83292.1586            8884247.51   
2012       91025.6970            8778214.89   
2013       95419.8056            8803373.36   
2014       99132.7393            8645590.71   
2015       95607.6604            8806403.56   
2016       98801.2564            8916152.41   

                                                summary  Month    Week  
Year                                                                    
2005                  Mostly cloudy throughout the day.     12      52  
2006  Partly cloudy throughout the day.Partly cloudy...  57168  232752  
2007  Partly cloudy starting in the morning continui...  57168  231528  
2008  Mostly cloudy starting in the morning.Mostly c...  57216  231552  
2009  Partly cloudy throughout the day.Partly cloudy...  57168  236520  
2010  Partly cloudy starting in the afternoon contin...  57360  233566  
2011  Mostly cloudy starting in the morning.Mostly c...  57168  232752  
2012  Partly cloudy until evening and breezy in the ...  57216  232776  
2013  Partly cloudy throughout the day.Partly cloudy...  57168  231528  
2014  Mostly cloudy throughout the day.Mostly cloudy...  57147  231443  
2015  Mostly cloudy throughout the day and breezy in...  57168  236520  
2016  Mostly cloudy throughout the day.Mostly cloudy...  57204  234020

In [39]:
years.sum(level='humidity').agg(['max'])

Summary           temp  \
max  Partly CloudyPartly CloudyPartly CloudyMostly ...  109709.688889   

          temp_app  humidity   wind_speed  Wind Bearing (degrees)  \
max  101823.322222   6999.63  100804.1769               1702875.0   

     Visibility (km)  Pressure (millibars)  \
max       99132.7393            8916152.41   

                                               summary  Month    Week  
max  Partly cloudy until evening and breezy in the ...  57360  236520

Самый влажный год - 2010

In [0]:
#weeks.dtypes.head(5)


date Summary weather_type     temp temp_app humidity  \
Week                                                                        
1     datetime64[ns, UTC]  object       object  float64  float64  float64   
2     datetime64[ns, UTC]  object       object  float64  float64  float64   
3     datetime64[ns, UTC]  object       object  float64  float64  float64   
4     datetime64[ns, UTC]  object       object  float64  float64  float64   
5     datetime64[ns, UTC]  object       object  float64  float64  float64   

     wind_speed Wind Bearing (degrees) Visibility (km) Pressure (millibars)  \
Week                                                                          
1       float64                float64         float64              float64   
2       float64                float64         float64              float64   
3       float64                float64         float64              float64   
4       float64                float64         float64              float64   
5       float64                float64         float64              float64   

     summary   Year  Month  
Week                        
1     object  int64  int64  
2     object  int64  int64  
3     object  int64  int64  
4     object  int64  int64  
5     object  int64  int64

In [40]:
weeks.temp.max()

Week
1     12.838889
2     15.738889
3     13.888889
4     14.772222
5     14.927778
6     16.111111
7     17.050000
8     17.266667
9     19.950000
10    19.883333
11    22.777778
12    22.344444
13    23.700000
14    26.111111
15    26.788889
16    26.205556
17    27.916667
18    31.111111
19    30.000000
20    30.000000
21    31.111111
22    33.816667
23    34.927778
24    33.911111
25    35.094444
26    36.200000
27    37.244444
28    36.205556
29    39.905556
30    38.700000
31    37.872222
32    37.822222
33    37.755556
34    38.861111
35    35.000000
36    37.150000
37    32.777778
38    35.950000
39    32.016667
40    27.733333
41    27.777778
42    27.588889
43    24.927778
44    23.866667
45    22.188889
46    21.105556
47    18.888889
48    17.222222
49    15.211111
50    11.066667
51    13.727778
52    16.094444
53    11.300000
Name: temp, dtype: float64

In [41]:
weeks.temp.min()

Week
1    -13.255556
2    -15.000000
3    -11.977778
4    -14.088889
5    -17.777778
6    -21.822222
7    -12.777778
8    -11.700000
9    -10.616667
10    -8.816667
11    -8.983333
12    -6.088889
13    -5.788889
14    -2.777778
15    -3.888889
16    -0.072222
17     0.138889
18     1.361111
19     3.738889
20     4.333333
21     8.011111
22     3.972222
23     5.577778
24     9.294444
25     9.083333
26     7.705556
27     9.516667
28    10.038889
29    11.466667
30    11.666667
31     9.800000
32     8.972222
33     8.761111
34     8.916667
35     7.711111
36     3.961111
37     4.472222
38     2.761111
39     2.894444
40    -2.783333
41    -2.288889
42    -6.061111
43    -2.311111
44    -6.044444
45    -5.944444
46    -7.244444
47    -4.866667
48    -6.161111
49   -13.094444
50   -15.483333
51   -16.666667
52   -16.666667
53   -10.111111
Name: temp, dtype: float64

In [42]:
t = (abs(weeks.temp.max()) - weeks.temp.mean()) + abs(weeks.temp.mean() - abs(weeks.temp.min()))
t

Week
1     25.916282
2     28.579299
3     23.291023
4     27.896266
5     32.641595
6     35.289544
7     25.089568
8     22.819462
9     21.739779
10    18.057684
11    18.146477
12    16.255556
13    17.911111
14    23.333333
15    22.900000
16    26.133333
17    27.777778
18    29.750000
19    26.261111
20    25.666667
21    23.100000
22    29.844444
23    29.350000
24    24.616667
25    26.011111
26    28.494444
27    27.727778
28    26.166667
29    28.438889
30    27.033333
31    28.072222
32    28.850000
33    28.994444
34    29.944444
35    27.288889
36    33.188889
37    28.305556
38    33.188889
39    29.122222
40    24.950000
41    25.488889
42    21.527778
43    22.616667
44    17.822222
45    16.244444
46    14.008868
47    14.022222
48    16.951052
49    19.586947
50    24.131560
51    28.774657
52    31.575301
53    19.959325
Name: temp, dtype: float64

In [43]:
t.agg(['max'])

max    35.289544
Name: temp, dtype: float64

По наибольшему оклонению от среднего получаем, что 6 неделя самая переменчивая